src/Microsoft.AspNet.Mvc.ViewFeatures/Controller.cs

```diff
@@ -131,17 +136,65 @@ public ActionContext ActionContext
         /// <summary>
         /// Gets or sets the <see cref="IModelMetadataProvider"/>.
         /// </summary>
-        [FromServices]
-        public IModelMetadataProvider MetadataProvider { get; set; }
+        public IModelMetadataProvider MetadataProvider
+        {
+            get
+            {
+                if (_metadataProvider == null)
+                {
+                    _metadataProvider = Resolver?.GetRequiredService<IModelMetadataProvider>();
+                }
+
+                return _metadataProvider;
+            }
+
+            [param: NotNull]
```

Comment:
I'd rather check this in before you get to the office and let your script deal with it :+1: